app/src/main/java/uk/co/nickbutcher/adaptiveiconplayground/AdaptiveIconView.kt

```diff
@@ -119,8 +119,12 @@ class AdaptiveIconView(
         background.eraseColor(Color.TRANSPARENT)
         foreground.eraseColor(Color.TRANSPARENT)
         val c = Canvas()
-        rasterize(icon.background, background, c)
-        rasterize(icon.foreground, foreground, c)
+        if(icon.background != null) {
```

Comment:
More idomatic kotlin might be:

`icon.background?.let { rasterize(it, background, c) }`